mz_bokeh_package/utilities/current_user.py

```diff
@@ -0,0 +1,96 @@
+import os
+from dataclasses import asdict
+from bokeh.io import curdoc
+
+from mz_bokeh_package.utilities.environment import Environment
+from mz_bokeh_package.utilities.graphql_api import MZGraphQLClient
+
+
+class CurrentUser:
+    """
+    Class with static methods for getting information about the current user
+    """
+
+    _users = {}
+
+    @staticmethod
+    def get_user_info(api_key: str | None) -> dict:
+        """Retrieves user information for the current session or API key, caching the information for future use.
+
+        Args:
+            api_key: unique user api key.
+
+        Returns:
+            dictionary of user "id" and "name".
+
+        Raises:
+             a ValueError if neither a session nor an API key is provided.
+        """
+
+        session_id = curdoc().session_context.id if curdoc().session_context else None
+
+        if session_id and session_id in CurrentUser._users:
+            return CurrentUser._users[session_id]
+
+        if session_id is None and api_key is None:
+            raise ValueError("api_key or an active session required to fetch user info.")
+
+        api_key = api_key or CurrentUser.get_api_key()
+
+        if not api_key:
+            raise ValueError("api_key or an active session required to fetch user info.")
+
+        user_info = asdict(MZGraphQLClient().get_user(api_key))
+
+        if session_id:
+            CurrentUser._users[session_id] = user_info
+            curdoc().on_session_destroyed(lambda: CurrentUser._remove_user_info(session_id))
+
+        return user_info
+
+    @staticmethod
+    def get_user_id() -> str | None:
+        """get the user_id of the current user
+
+        Returns:
+            the user_id of the current user
+        """
+
+        return MZGraphQLClient().get_user(CurrentUser.get_api_key()).id
```

Comment:
Here and in `get_user_name` you were supposed to call `get_user_info` (that was the whole point :) ).